app/src/common/shared/org/mozilla/vrbrowser/browser/adapter/ComponentsAdapter.kt

```diff
@@ -0,0 +1,68 @@
+package org.mozilla.vrbrowser.browser.adapter
+
+import mozilla.components.browser.state.action.EngineAction
+import mozilla.components.browser.state.action.TabListAction
+import mozilla.components.browser.state.state.*
+import mozilla.components.browser.state.store.BrowserStore
+import org.mozilla.geckoview.GeckoSession
+import org.mozilla.vrbrowser.browser.components.GeckoEngineSession
+import org.mozilla.vrbrowser.browser.engine.Session
+
+class ComponentsAdapter private constructor(
+        val store: BrowserStore = BrowserStore()
+) {
+    companion object {
+        private val instance: ComponentsAdapter = ComponentsAdapter()
+
+        fun get(): ComponentsAdapter = instance
```

Comment:
nit: You could consider using `@JvmStatic` here so that you can use `ComponentsAdapter.get()` directly from Java.